src/RoadRegistry.Wms.ProjectionHost/EventProcessor.cs

```diff
@@ -59,7 +57,9 @@ public class EventProcessor : IHostedService
             Func<WmsContext> dbContextFactory,
             IStreetNameCache streetNameCache,
             Scheduler scheduler,
-            ILogger<EventProcessor> logger)
+            ILogger<EventProcessor> logger,
+            IHostApplicationLifetime lifeTime,
```

Comment:
Using this to end the application is _weird_ ... I'm also wondering if and when the synchronization with the cache will run - you can't interleave calls, so even if the message gets scheduled, it will never be picked up, because by the time it would get picked up, the resume message will have submitted the request to end the application. It's clear that the current design of the event processor does not fit the use case. One option, which would fit better with the current design, would be to "internalize" rather than "externalize" the schedule. That is, keep the container running, configure the intervals at which it should do a catchup and update metadata (using the existing scheduler). Alternatively, if you feel stronger about externalizing (making it an infra concern by running the container on regular intervals), you could make the code more procedural and convert each of the internal messages the eventprocessor handles to methods you can call from `Program.cs`, thereby not needing an `IHostApplicationLifetime` to begin with and even the IMetadataUpdater could be just a dependency of `Program.cs` rather than `EventProcessor.cs`. Anyway, just my 2c.

Reply:
It is weird. Guess I was a bit too focused on making it work. I'll rework it to be more procedural, I feel better about externalizing the schedule.